### DISCUSSION TITLE
Extending a Module in V2

### DISCUSSION THREAD

**Discussion by fxmb** [medusajs/medusa#9484] [Q&A]
To me it is not clear how to extend a module in V2. I went through the docs multiple times, asked for help on discord and there seem to be many confusions for many devs around this. The docs are either wrong or not complete as it seems. 

In particular:
1. I cannot pass additional_data to core endpoints (field is not allowed by validator)
2. I do not understand how I can adjust a core endpoint to return its extended fields by default. 

Here is the existing discord discussion:

 https://discord.com/channels/876835651130097704/1287715752471822377

**Comment by fxmb** [comment]
This gets a lot of upvotes apparently... does anyone know this? Its crucial for us and I feel like no one from the medusa team is responding right now (I know they are busy preparing V2). Thanks for any pointeres

**Reply by fxmb** [reply]
Anyone?

**Comment by PiotrKukuc12** [comment]
I am having the same issue.

**Comment by AlexAntonides** [comment]
I'm experiencing the same right now. I'm trying to extend the LineItem of an Order to add my own properties to it, and I thought it  [extending the data modules](https://docs.medusajs.com/resources/commerce-modules/cart/extend) would allow me to do this, but after hours of struggling, we came to the conclusion that this is not possible. 
I'm really disappointed in the extendability of the core models and it is crucial for us to extend these entities. Especially now, even after the v2 release, the lack of response from the dev team is quite a bummer.

**Comment by dawidstolarski** [comment]
Anything happened here? Issue is critical IMO, time goes by and we don't have any reply from you guys...

**Comment by DevKevvv** [comment]
I think we can only use the “metadata” field in the entities for this... But even here, not all entities have this field.
